### DISCUSSION TITLE
problemas compilando - TP i

### DISCUSSION THREAD

**Discussion by rinrot** [INGCOM-UNRN-P1/cursada-2023#230] [Preguntas y respuestas] [answered]
Hola! estaba separando las funciones para tenerlas un poco mas ordenadas, y cuando me puse a compilar me saltaban errores del struct del jugador.

![imagen](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/134400606/2dc16b60-5eb9-4059-8064-3fdbdb04af7a)

que tengo que hacer para que funque?

**Comment by TeoInMaghine** [comment]
Si no me equivoco esto es porque estás definiendo `jugador` y `jugador_t` varias veces. Creo que hay varias formas de arreglarlo, tal vez reordenando un poco el código, pero una forma es meter esa definición en un archivo .h (creo que con un .c también funciona), meter un `#include` de ese archivo donde sea que uses `jugador` y `jugador_t`, y usar `#ifndef` en el .h para que no se defina más de una vez. Ejemplo:

En un archivo jugador.h:
```C
// Si JUGADOR no está definido, definirlo
#ifndef JUGADOR
#define JUGADOR

// Acá adentro podés poner las estructuras, macros y librerías que quieras definir
#include <stdio.h>
#define TRUE 1
#define FALSE 0

typedef struct jugador {
    ...
} jugador_t; 

#endif
```

Por lo que veo hubo otra pregunta relacionada antes: #186. Realmente no sé si es sobre-complicarla hacerlo así, como dije tal vez ordenar el código de otra forma resuelve tu problema y aparte te deja un código más legible, como no tengo acceso a tu repo no tengo idea realmente pero tal vez te sirve esta forma.

**Reply by martinvilu** [reply]
Mandá las estructuras y los define a este `.h` que siguiere Teo.

La _verdadera_ separación en modulos, complica la compilación

**Reply by rinrot** [reply]
cuando armo jugador.h el include detecta errores
![imagen](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/134400606/9c01931c-da6e-44e3-89c1-6206a977f8ae)

los contenidos de jugador.h son estos:
```
#ifndef JUGADOR
#define JUGADOR

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#define TRUE 1
#define FALSE 0

typedef struct jugador {
    int INT;
    int PSI;
    int FYS;
    int MOT;
    
    int salud;
    int moral;
} jugador_t;

#endif
```

tengo tantos problemas y errores que no voy ni 2/3 del trabajo

**Reply by TeoInMaghine** [reply, marked as answer]
Sería `#include "jugador.h"`, con `"` en vez de `<`.

Importante a tener en cuenta que tiene que estar en la misma carpeta, si jugador.h está en una carpeta "más arriba" podés poner `#include "../jugador.h"`, y si está en una subcarpeta podés poner `#include "subcarpeta/jugador.h"`, no sé si me explico.

Ah y si no necesitás el TRUE y el FALSE no hace falta que los pongas, yo los puse como ejemplo de que podés definir macros ahí adentro.

**Reply by rinrot** [reply]
el true y el false los estaba usando para verificar status. ahora que lo cambie funciona. Muchisimas gracias!
